validator/client/runner.go

```diff
@@ -121,7 +59,9 @@ func run(ctx context.Context, v iface.Validator) {
 		log.Fatalf("Failed to update proposer settings: %v", err) // allow fatal. skipcq
 	}
 	for {
-		slotCtx, cancel := context.WithCancel(ctx)
+		var slotCtx context.Context
+		var cancel context.CancelFunc
+		slotCtx, cancel = context.WithCancel(ctx)
```

Comment:
Does this change make a difference?

Reply:
Thanks. The original way (after refactor) will complain "ineffective assignment" error. The reason is the `slotCtx` variable is assigned too early and it will soon be reassigned. I changed it a little bit to escape from this error.